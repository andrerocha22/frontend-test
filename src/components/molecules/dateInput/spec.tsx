import React from 'react';
import { shallow, ShallowWrapper, mount } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import 'jest-styled-components';

import DateInput from './index';

const defaultProps = {
  iconSrcLeft: 'iconTest',
  iconSrcRight: 'iconTest',
};

const mockCallback = jest.fn();

describe('test date input', () => {
  const setup = (props = {}) => {
    const setupProps = { ...defaultProps, ...props };
    return shallow(<DateInput diffCallback={mockCallback} {...setupProps} />);
  };

  const setupMount = (props = {}) => {
    const setupProps = { ...defaultProps, ...props };
    return mount(<DateInput diffCallback={mockCallback} {...setupProps} />);
  };

  let component: ShallowWrapper = undefined;

  test('renders without error', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-date-input');
    expect(component.length).toBe(1);
  });

  test('date input snapshot test ', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-date-input');
    expect(component).toMatchSnapshot();
  });

  test('date input test props', () => {
    const wrapper = setupMount();
    expect(wrapper.prop('iconSrcLeft')).toEqual(defaultProps.iconSrcLeft);
  });

  test('date input test props', () => {
    const wrapper = setupMount();
    expect(wrapper.prop('iconSrcRight')).toEqual(defaultProps.iconSrcRight);
  });

  // test('date input test props', () => {
  //   const wrapper = setupMount();
  //   const buttonForward = wrapper.find('button');
  //   buttonForward.simulate('click');
  //   const spy = jest.spyOn(DateInput.propTypes, "")

  //   expect(wrapper.)
  // });
});
