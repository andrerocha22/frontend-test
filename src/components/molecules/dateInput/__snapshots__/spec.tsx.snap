// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test date input date input snapshot test 1`] = `
<styled.div
  data-test="component-date-input"
>
  <ButtonSelection
    backgroundColor="#E1E8ED"
    disabled={false}
    handleClick={[Function]}
    onKeyUp={[Function]}
  >
    <img
      src="iconTest"
    />
  </ButtonSelection>
  <styled.span
    data-test="component-date-display"
  >
    <styled.span
      className="month"
      data-test="component-month-text"
    >
      October
    </styled.span>
    <styled.span
      className="year"
      data-test="component-year-text"
    >
      2020
    </styled.span>
  </styled.span>
  <ButtonSelection
    backgroundColor="#E1E8ED"
    handleClick={[Function]}
    onKeyUp={[Function]}
  >
    <img
      src="iconTest"
    />
  </ButtonSelection>
</styled.div>
`;
